api/types/types.go

```diff
@@ -461,13 +461,23 @@ type Runtime struct {
 	Args []string `json:"runtimeArgs,omitempty"`
 }
 
+// DownloadCacheUsage contains information about the disk usage of the download
+// cache
+type DownloadCacheUsage struct {
+	TotalNumber  int64
+	TotalBytes   int64
+	ActiveNumber int64
+	ActiveBytes  int64
+}
+
 // DiskUsage contains response of Engine API:
 // GET "/system/df"
 type DiskUsage struct {
-	LayersSize int64
-	Images     []*ImageSummary
-	Containers []*Container
-	Volumes    []*Volume
+	LayersSize         int64
+	Images             []*ImageSummary
+	DownloadCacheUsage DownloadCacheUsage
```

Comment:
Grepping the docker code base shows that there may be a few other caches in docker:

```
$ git grep cache | wc -l
771
```

Looks like mostly for builder, other image related functions and some graphdriver stuff.

Placing this into a slice with a name field is not a premature generalization. We need to avoid creating implementations that prevent generalization in the future.